index.bs

```diff
@@ -6913,54 +6755,93 @@ may return any value, which will be discarded.
 </p>
 
 
-<h4 id="es-unsigned-long-long">unsigned long long</h4>
+<h5 id="es-unsigned-long-long">unsigned long long</h5>
 
 <div id="es-to-unsigned-long-long" algorithm="convert an ECMAScript value to unsigned long long">
 
     An ECMAScript value |V| is [=converted to an IDL value|converted=]
     to an IDL {{unsigned long long}} value by running the following algorithm:
 
-    1.  Initialize |x| to [=ToNumber=](|V|).
+    1.  Let |x| be [=?=] <a abstract-op>ConvertToInt</a>(|V|, 64, "unsigned").
+    1.  Return the IDL {{long}} value that represents the same numeric value as |x|.
+</div>
+
+<p id="unsigned-long-long-to-es">
+    The result of [=converted to an ECMAScript value|converting=]
+    an IDL {{unsigned long long}} value to an ECMAScript
+    value is a <emu-val>Number</emu-val> value that
+    represents the closest numeric value to the {{unsigned long long}},
+    choosing the numeric value with an <em>even significand</em> if there are
+    two [=equally close values=].
+    If the {{unsigned long long}} is less than or equal to 2<sup>53</sup> − 1,
+    then the <emu-val>Number</emu-val> will be able to
+    represent exactly the same value as the
+    {{unsigned long long}}.
+</p>
+
+<h5 id="es-integer-types-abstract-ops">Abstract operations</h5>
+
+<div algorithm>
+
+    <dfn lt="IntegerPart" abstract-op>IntegerPart(|n|)</dfn>:
+
+    1.  Let |r| be [=floor=]([=abs=](|n|))
+    1.  If |n| &lt; 0, then return -1 × |r|.
+    1.  Otherwise, return |r|.
+</div>
+
+<div algorithm>
+    <dfn lt="ConvertToInt" abstract-op>ConvertToInt(|V|, |bitLength|, |signedness|)</dfn>:
```

Comment:
Fixed in https://github.com/heycam/webidl/commit/36168b74d698015c015d94f2a27414795292217b